backend/src/models/entry/entryAnalysis.ts

```diff
@@ -85,64 +73,36 @@ async function removeLegacyApiKey(configId: string): Promise<string> {
       }
     }
   }
-  return config.model.analysis;
-}
 
-async function getAnalysisCompletion(
-  configModelAnalysis: string,
-  content: string
-): Promise<any> {
-  if (process.env.OPENAI_API_KEY === undefined) {
-    throw new Error('OPENAI_API_KEY undefined. Must be set to get analysis');
+  if (!process.env.OPENAI_API_KEY) {
+    throw new Error('API key is not available)');
   }
-  const cdGpt = new CdGpt(
-    process.env.OPENAI_API_KEY,
-    configModelAnalysis,
-    '',
-    0.7
-  );
+
+  const cdGpt = new CdGpt(process.env.OPENAI_API_KEY, config.model.analysis);
 
   cdGpt.seedAnalysisMessages();
   cdGpt.addUserMessage({ analysis: content });
 
-  const analysisCompletion: any = await cdGpt.getAnalysisCompletion();
+  const analysisCompletion = await cdGpt.getAnalysisCompletion();
 
   if (analysisCompletion.error) {
     throw new Error(analysisCompletion.error.message);
   }
 
-  return JSON.parse(analysisCompletion.choices[0].message.content);
-}
+  const response = JSON.parse(analysisCompletion.choices[0].message.content);
 
-// Get the analysis content for an entry
-entryAnalysisSchema.methods.getAnalysisContent = async function (
-  configId: string,
-  content: string
-): Promise<any> {
-  const configModelAnalysis = await removeLegacyApiKey(configId);
-  const response = await getAnalysisCompletion(configModelAnalysis, content);
-  const {
```

Comment:
What are your vscode linter settings? Could I get a copy of them? I copied the one you messaged me, but it's autoformatting like this on save. It seems like this style isn't what you're looking for.

Reply:
I don't have any custom user or workspace linter settings. vscode is using the eslint config to highlight errors and prettier is applying those rules on those that can be formatted on save. But if it's not setup correctly vscode could be pulling rules from other configurations. I think prettier comes with its own and maybe some other language specific plugins do. So you need to play around with your settings a bit to make sure vscode is using eslint config.